### DISCUSSION TITLE
Pretty Printer for lldb

### DISCUSSION THREAD

**Discussion by Finkman** [nlohmann/json#4184] [Q&A]
We'd like to move from gdb to lldb, since it has some advantages for us.
Is there a way to enable pretty printing on debugging sessions?
We used the python script for gdb and would like to know if there is something similar on lldb.
Thx.

**Comment by nlohmann** [comment]
There is currently no bundled LLDB pretty printer in this repository — only a GDB Python pretty printer exists, under tools/gdb_pretty_printer. LLDB can load GDB-style printers in some setups, but there's no first-party LLDB script/formatter here, so you'd need to port the existing GDB printer to LLDB's Python scripting API (lldb.formatters) yourself or check for a community-maintained one.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
